app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -664,6 +666,52 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
         }.launch(intent)
     }
 
+    /**
+     * Prompt the user with a new category name to assign,
+     * if the new category forms/leads a path (i.e. contains "/"), intermediate directories will be created
+     * and new category will be placed inside.
+     *
+     * @param oldCategory The category to change its name
+     * @param error Determines whether to show an error to the user in the alert dialog, this error
+     * may be due to the new category the user entered already exists or the field was empty
+     */
+    private fun renameCategory(oldCategory: PasswordItem, error: Boolean = false) {
+        val view = layoutInflater.inflate(R.layout.folder_dialog_fragment, null)
+        val newCategoryEditText = view.findViewById<TextInputEditText>(R.id.folder_name_text)
+        if (error) {
+            newCategoryEditText.error = getString(R.string.message_error_rename_folder)
+        }
+
+        val dialog = MaterialAlertDialogBuilder(this)
+            .setTitle(R.string.title_rename_folder)
+            .setView(view)
+            .setMessage(getString(R.string.message_rename_folder, oldCategory.name))
+            .setPositiveButton(R.string.dialog_ok) { _, _ ->
+                val newCategory = File("${oldCategory.file.parent}/${newCategoryEditText.text}")
+                when {
+                    newCategoryEditText.text.isNullOrBlank() ||
+                        newCategory.exists() -> renameCategory(oldCategory, true)
+                    else -> {
+                        newCategory.mkdirs()
+                        oldCategory.file.renameTo(newCategory)
+                        AutofillMatcher.updateMatches(this, mapOf(oldCategory.file to newCategory))
```

Comment:
`AutofillMatcher.updateMatches` currently expects a (recursive) list of all files and their new locations, not just a containing folder. I will try to add a variant that supports this.

Reply:
I have reused the existing move code, but please test this.